packages/backend-core/src/accounts/accounts.ts

```diff
@@ -5,9 +5,14 @@ import { CloudAccount } from "@budibase/types"
 
 const api = new API(env.ACCOUNT_PORTAL_URL)
 
+const EXIT_EARLY = env.SELF_HOSTED || env.DISABLE_ACCOUNT_PORTAL
```

Comment:
What is the logic behind this exit early. It might be confusing, as any consumer of the service will not be aware of these functions returning always undefined. This "undefined" seems more of a business requirement, could it be moved to a higher implementation level?

Reply:
This is a cloud vs self hosted consideration. We only want to use this client in the case of being cloud deployed. 
The current usages always have to wrap the call with their own checks. e.g:
- https://github.com/Budibase/budibase/blob/1dc3d44a44e5207c63cc1ec6a8e3ea247e5d0765/packages/backend-core/src/cache/user.ts#L17
- https://github.com/Budibase/budibase/blob/8d4b59126afc99cebad4bfcafa3faf923f7d51a0/packages/worker/src/sdk/users/users.ts#L567
-  https://github.com/Budibase/budibase/blob/8d4b59126afc99cebad4bfcafa3faf923f7d51a0/packages/worker/src/sdk/users/users.ts#L188


I've added this check internally now so everything is centralised here instead. 

The type signature for `getAccount` and `getAccountByTenantId` already tell us undefined is an potential response, I've updated `getStatus` as well to show the response format and the potential for undefined. 

see: https://github.com/Budibase/budibase/pull/9672/commits/6750a035b96510c586d796301ace93c2a32a7365